community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/valueHelper.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compatibility
+
+import java.util.function.BiConsumer
+
+import org.neo4j.values.AnyValue
+import org.neo4j.values.storable.{BooleanValue, FloatingPointValue, LongValue, TextValue, NoValue}
+import org.neo4j.values.virtual.EdgeValue.RelationshipProxyWrappingEdgeValue
+import org.neo4j.values.virtual.{ListValue, MapValue}
+import org.neo4j.values.virtual.NodeValue.NodeProxyWrappingNodeValue
+
+import scala.collection.mutable
+
+object valueHelper {
+  def fromValue(value: AnyValue): Any = value match {
```

Comment:
This should probably use an `AnyValueWriter` instead of doing type matching. See `asObject` in `TransactionalQueryContext` in 3.3